benchmarks/report_bayesmark.py

```diff
@@ -1,55 +1,255 @@
-from collections import Counter
+from abc import ABC
+from collections import defaultdict
 import io
-import json
+import itertools
 import os
 from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
 
+import numpy as np
+import pandas as pd
+from scipy.special import binom
+from scipy.stats import mannwhitneyu
 
-_TABLE_HEADER = "\n|Ranking|Solver|Score|\n|---:|:---|---:|\n"
+
+_LINE_BREAK = "\n"
+_TABLE_HEADER = "|Ranking|Solver|"
+_HEADER_FORMAT = "|:---|---:|"
+_OVERALL_HEADER = "|Solver|Borda|Firsts|\n|:---|---:|---:|\n"
+
+Moments = Tuple[float, float]
+Samples = Dict[str, List[float]]
+
+
+class BaseMetric(ABC):
+    @property
+    def name(self) -> str:
+        """Metric name displayed in final report."""
+
+    def calculate(self, data: pd.DataFrame) -> List[float]:
+        """Calculates metric for each study in data frame."""
```

Comment:
It is recommended to write it in the same way as Optuna's other abstract classes.
```suggestion
class BaseMetric(object, metaclass=abc.ABCMeta):
    @abc.abstractmethod
    @property
    def name(self) -> str:
        """Metric name displayed in final report."""

    @abc.abstractmethod
    def calculate(self, data: pd.DataFrame) -> List[float]:
        """Calculates metric for each study in data frame."""
```